src/cascadia/TerminalSettingsEditor/LaunchViewModel.cpp

```diff
@@ -62,4 +57,30 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 
         return winrt::single_threaded_observable_vector(std::move(profiles));
     }
+
+    winrt::Windows::Foundation::IInspectable LaunchViewModel::CurrentDefaultTerminal()
+    {
+        return winrt::box_value(_Settings.CurrentDefaultTerminal());
+    }
+
+    void LaunchViewModel::CurrentDefaultTerminal(const IInspectable& value)
+    {
+        const auto defaultTerminal{ winrt::unbox_value<Model::DefaultTerminal>(value) };
+        _Settings.CurrentDefaultTerminal(defaultTerminal);
+    }
+
+    winrt::Windows::Foundation::Collections::IObservableVector<winrt::Windows::Foundation::IInspectable> LaunchViewModel::DefaultTerminals() const
+    {
+        const auto allTerminals = _Settings.DefaultTerminals();
+
+        std::vector<IInspectable> terminals;
+        terminals.reserve(allTerminals.Size());
+
+        for (const auto& terminal : allTerminals)
+        {
+            terminals.emplace_back(terminal);
+        }
+
+        return winrt::single_threaded_observable_vector(std::move(terminals));

```

Comment:
There really should be a better way to do this?

Reply:
Can't this just return `_Settings.DefaultTerminals()`? We shouldn't need to make a whole copy of the list here... right?

Reply:
I was thrown off by `DefaultProfiles` returning a vector of IInspectable - fixed them both now!